packages/eas-cli/src/commands/update/configure.ts

```diff
@@ -242,6 +249,36 @@ async function configureAppJSONForEASUpdateAsync({
   return result.config.expo;
 }
 
+async function checkAndConfigureEasJsonReleaseChannelsAsync(projectDir: string): Promise<boolean> {
+  const reader = new EasJsonReader(projectDir);
+  const easJson = await reader.readAsync();
+
+  let releaseChannelsFound = false;
+
+  if (easJson.build) {
+    for (const [key, value] of Object.entries(easJson.build)) {
+      if (value.releaseChannel) {
+        releaseChannelsFound = true;
+        easJson.build[key].channel = value.releaseChannel;
+        delete easJson.build[key].releaseChannel;
+      }
+    }
+  }
+
+  if (releaseChannelsFound) {
+    try {
+      await fs.writeFile(
```

Comment:
@wkozyra95 
- You can't use both at the same time, because the `releaseChannel` and `channel` fields are mutually exclusive
- We modify and write app.json as well
- If someone really doesn't like the changes they can decide to rollback the change and not commit it

But most importantly: I think the amount of people missing this migration change in the docs and ending up with updates that don't work outweighs any downsides for the tiny minority who has manually applied some special indentation to the file.